java/ql/src/semmle/code/java/dataflow/FlowSources.qll

```diff
@@ -298,7 +298,8 @@ private class SpringWebRequestGetMethod extends Method {
   }
 }
 
-private class EnvTaintedMethod extends Method {
+/** Models methods that are tainted by the environment of the user, such as `System.getProperty` or `System.getenv()`. */
+class EnvTaintedMethod extends Method {
```

Comment:
I'd prefer to keep this private (if it is exposed into a broader context, then I don't think its name fits very well anymore). In particular, you're only using parts of its extent and not the `MethodSystemGetenv` part, so I think it's better to just refer to the two other parts (`PropertiesGetPropertyMethod` and `MethodSystemGetProperty`) directly.

Reply:
Can you clarify what you mean with this:
> you're only using parts of its extent and not the MethodSystemGetenv part

I'm using `EnvTaintedMethod` to model these cases:
```java
System.getProperty("disableSSL");
System.getenv("disableSSL");
System.getProperties().getProperty("disableSSL");
```

Reply:
Ah, right. I was thinking about `System.getenv()` and missed `System.getenv(String)`. In that case we should rename the class so it doesn't talk about taint, since we're now exposing it to other contexts.
```suggestion
/** A method that reads from the environment, such as `System.getProperty` or `System.getenv`. */
class EnvReadMethod extends Method {
```

Reply:
Done!